SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestLoadBalancingData/SpatialTestLoadBalancingData.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDK/SpatialTestLoadBalancingData/SpatialTestLoadBalancingData.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+
+#include "SpatialCommonTypes.h"
+#include "SpatialConstants.h"
+#include "Test1x2GridStrategy.h"
+
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "Kismet/GameplayStatics.h"
+
+void USpatialTestLoadBalancingDataTestMap::CreateCustomContentForMap()
+{
+	Super::CreateCustomContentForMap();
+
+	Cast<ASpatialWorldSettings>(World->GetWorldSettings())
+		->SetMultiWorkerSettingsClass(USpatialTestLoadBalancingDataMultiWorkerSettings::StaticClass());
+
+	AddActorToLevel<ASpatialTestLoadBalancingData>(World->GetCurrentLevel(), FTransform::Identity);
+}
+
+USpatialTestLoadBalancingDataMultiWorkerSettings::USpatialTestLoadBalancingDataMultiWorkerSettings()
+{
+	WorkerLayers[0].ActorClasses.Add(ASpatialTestLoadBalancingDataActor::StaticClass());
+	WorkerLayers.Add(
+		{ TEXT("Offloaded"), { ASpatialTestLoadBalancingDataOffloadedActor::StaticClass() }, USingleWorkerStrategy::StaticClass() });
+	WorkerLayers.Add({ TEXT("Grid"), { ASpatialTestLoadBalancingDataZonedActor::StaticClass() }, UTest1x2GridStrategy::StaticClass() });
+}
+
+ASpatialTestLoadBalancingDataActor::ASpatialTestLoadBalancingDataActor()
+{
+	bReplicates = true;
+
+	bAlwaysRelevant = true;
+}
+
+ASpatialTestLoadBalancingDataOffloadedActor::ASpatialTestLoadBalancingDataOffloadedActor()
+{
+	bReplicates = true;
+
+	bAlwaysRelevant = true;
+}
+
+ASpatialTestLoadBalancingDataZonedActor::ASpatialTestLoadBalancingDataZonedActor()
+{
+	bReplicates = true;
+
+	bAlwaysRelevant = true;
+
+	RootComponent = CreateDefaultSubobject<USceneComponent>(TEXT("RootComponent"));
+}
+
+template <class T>
+T* GetWorldActor(UWorld* World)
+{
+	TArray<AActor*> DiscoveredActors;
+	UGameplayStatics::GetAllActorsOfClass(World, T::StaticClass(), DiscoveredActors);
+	if (DiscoveredActors.Num() == 1)
+	{
+		return Cast<T>(DiscoveredActors[0]);
+	}
+	return nullptr;
+}
+
+template <class T, int Count, class U>
+void GetWorldActors(UWorld* World, TArray<U>& OutActors)
+{
+	TArray<AActor*> DiscoveredActors;
+	UGameplayStatics::GetAllActorsOfClass(World, T::StaticClass(), DiscoveredActors);
+	if (DiscoveredActors.Num() == Count)
+	{
+		DiscoveredActors.Sort([](const AActor& Lhs, const AActor& Rhs) {
+			return Lhs.GetActorLocation().Y < Rhs.GetActorLocation().Y;
+		});
+		Algo::Transform(DiscoveredActors, OutActors, [](AActor* Actor) -> U {
+			return Cast<T>(Actor);
+		});
+	}
+}
+
+void ASpatialTestLoadBalancingData::PrepareTest()
+{
+	Super::PrepareTest();
+
```

Comment:
Could you add a comment defining the expected lbs here, and how layers map to server id numbers